pkg/component/watch_test.go

```diff
@@ -233,42 +227,37 @@ func TestWatchAndPush(t *testing.T) {
 			go func() {
 				t.Logf("Starting file simulations \n%+v\n", tt.fileModifications)
 				// Simulating file modifications for watch to observe
-				for {
-					select {
-					case startMsg := <-StartChan:
-						if startMsg {
-							for _, fileModification := range tt.fileModifications {
+				startMsg := <-StartChan
+				if startMsg {
+					for _, fileModification := range tt.fileModifications {
```

Comment:
@kadel Please have a look at this too

Reply:
I think that the `for` loop is needed, as it is being used to wait for the data from the `StartChan`